edge-apps/rss-reader/static/js/main.js

```diff
@@ -133,3 +131,191 @@ const getRssData = function () {
+document.addEventListener('rssDataLoaded', async () => {
+  const { getNearestCity } = OfflineGeocodeCity
+  const allTimezones = moment.tz.names()
+
+  const sentryDsn = screenly.settings.sentry_dsn
+  // Initiate Sentry.
+  if (sentryDsn) {
+    Sentry.init({
+      dsn: sentryDsn
+    })
+  } else {
+    console.warn('Sentry DSN is not defined. Sentry will not be initialized.')
+  }
+
+  async function timeAndLocale () {
+    const { metadata, settings } = screenly
+    const latitude = metadata.coordinates[0]
+    const longitude = metadata.coordinates[1]
+    const defaultLocale = navigator?.languages?.length
+      ? navigator.languages[0]
+      : navigator.language
+
+    const getLocale = async () => {
+      const overrideLocale = settings?.override_locale || 'en'
+
+      if (overrideLocale) {
+        if (moment.locales().includes(overrideLocale)) {
+          return overrideLocale
+        } else {
+          console.warn(`Invalid locale: ${overrideLocale}. Using defaults.`)
+        }
+      }
+
+      const data = await getNearestCity(latitude, longitude)
+      const countryCode = data.countryIso2.toUpperCase()
+      return clm.getLocaleByAlpha2(countryCode) || defaultLocale
+    }
+
+    const getTimezone = async () => {
+      const overrideTimezone = settings?.override_timezone
+      if (overrideTimezone) {
+        if (allTimezones.includes(overrideTimezone)) {
+          return overrideTimezone
+        } else {
+          console.warn(`Invalid timezone: ${overrideTimezone}. Using defaults.`)
+        }
+      }
+
+      return tzlookup(latitude, longitude)
+    }
+
+    const initDateTime = async () => {
+      const timezone = await getTimezone()
+      const locale = await getLocale()
+      const momentObject = moment().tz(timezone)
+
+      if (locale) {
+        momentObject.locale(locale)
+      }
+      const dayOfMonth = momentObject.format('DD')
+      document.querySelector('.date-text').innerText = momentObject.format('ddd').toUpperCase()
+      document.querySelector('.date-number').innerText = dayOfMonth
+
+      const format = moment.localeData(locale).longDateFormat('LT').includes('A') ? 'ddd MMM DD YYYY hh:mm A' : 'ddd MMM DD YYYY HH:mm'
+      const pubDate0 = moment(localStorage.getItem('pubDate-0')).tz(timezone).locale(locale).format(format)
+      const pubDate1 = moment(localStorage.getItem('pubDate-1')).tz(timezone).locale(locale).format(format)
+      const pubDate2 = moment(localStorage.getItem('pubDate-2')).tz(timezone).locale(locale).format(format)
+      const pubDate3 = moment(localStorage.getItem('pubDate-3')).tz(timezone).locale(locale).format(format)
+
+      document.querySelector('.content-timestamp-0').innerText = pubDate0
+      document.querySelector('.content-timestamp-1').innerText = pubDate1
+      document.querySelector('.content-timestamp-2').innerText = pubDate2
+      document.querySelector('.content-timestamp-3').innerText = pubDate3
+    }
```

Comment:
This can be generalized. A loop can be used.

Reply:
Thanks, Using a loop now. 

```
  const format = moment.localeData(locale).longDateFormat('LT').includes('A') ? 'ddd MMM DD YYYY hh:mm A' : 'ddd MMM DD YYYY HH:mm'
  const pubDates = ['pubDate-0', 'pubDate-1', 'pubDate-2', 'pubDate-3'].map(dateKey => {
        return moment(localStorage.getItem(dateKey)).tz(timezone).locale(locale).format(format)
      })

 pubDates.forEach((pubDate, index) => {
        document.querySelector(`.content-timestamp-${index}`).innerText = pubDate
      })
```